src/Symfony/Component/TypeInfo/Type/BuiltinType.php

```diff
@@ -41,15 +41,10 @@ public function getTypeIdentifier(): TypeIdentifier
     public function isIdentifiedBy(TypeIdentifier|string ...$identifiers): bool
     {
         foreach ($identifiers as $identifier) {
-            if (\is_string($identifier)) {
-                try {
-                    $identifier = TypeIdentifier::from($identifier);
-                } catch (\ValueError) {
-                    continue;
-                }
-            }
-
-            if ($identifier === $this->typeIdentifier) {
+            if (
+                $identifier === $this->typeIdentifier
+                || $identifier === $this->typeIdentifier->value
+            ) {
```

Comment:
thank you for this PR, simplicity FTW!

```suggestion
            if ($identifier === $this->typeIdentifier || $identifier === $this->typeIdentifier->value) {
```

Reply:
fixed.